src/nvim/getchar.c

```diff
@@ -2250,23 +2250,26 @@ static int get_key_from_user(int *timedoutp, int *mode_deletedp,
       setcursor();                /* put cursor back where it belongs */
   }
 
-  if (bytes_read < 0)
-    return -1;                     /* end of input script reached */
-  if (bytes_read == NUL) {                 /* no character available */
+  if (bytes_read < 0) {
+    /* end of input script reached will return -1 below */
+  } else if (bytes_read == NUL) {
```

Comment:
Wait, `bytes_read` is supposed to represent a character?! That's not a very good variable name...